fe/fe-core/src/main/java/org/apache/doris/nereids/rules/implementation/LogicalWindowToPhysicalWindow.java

```diff
@@ -0,0 +1,539 @@
+
+        // PhysicalWindow nodes for each different window frame, so at least one PhysicalWindow node will be added
+        for (WindowFrameGroup windowFrameGroup : orderKeyGroup.groupList) {
+            newRoot = createPhysicalWindow(newRoot, windowFrameGroup, logicalWindow, requiredOrderKeys);
+        }
+
+        return newRoot;
+    }
+
+    private List<OrderKey> generateKeysNeedToBeSorted(OrderKeyGroup orderKeyGroup) {
+        // all keys that need to be sorted, which includes BOTH partitionKeys and orderKeys from this group
+        List<OrderKey> keysNeedToBeSortedList = Lists.newArrayList();
+
+        // used as SortNode.isAnalyticSort, but it is unnecessary to add it in LogicalSort
+        if (!orderKeyGroup.partitionKeyList.isEmpty()) {
+            keysNeedToBeSortedList.addAll(orderKeyGroup.partitionKeyList.stream().map(partitionKey -> {
+                // todo: haven't support isNullFirst, and its default value is false(see AnalyticPlanner#line403,
+                //  but in LogicalPlanBuilder, its default value is true)
+                return new OrderKey(partitionKey, true, false);
+            }).collect(Collectors.toList()));
+        }
+
+        if (!orderKeyGroup.orderKeyList.isEmpty()) {
+            keysNeedToBeSortedList.addAll(orderKeyGroup.orderKeyList);
+        }
+        return keysNeedToBeSortedList;
+    }
+
+    private PhysicalWindow createPhysicalWindow(Plan root, WindowFrameGroup windowFrameGroup,
+                                                LogicalWindow logicalWindow, List<OrderKey> requiredOrderKeys) {
+        // todo: partitionByEq and orderByEq?
+        // requiredProperties:
+        // Distribution: partitionKeys
+        // Order: requiredOrderKeys
+        PhysicalWindow physicalWindow = new PhysicalWindow<>(
+                logicalWindow.getOutputExpressions(),
+                logicalWindow.getWindowExpressions(),
+                windowFrameGroup,
+                logicalWindow.getLogicalProperties(),
+                root);
+        // todo: add isAnalyticSort to physicalWindow
+        if (windowFrameGroup.partitionKeyList.isEmpty() && requiredOrderKeys.isEmpty()) {
+            return physicalWindow.withRequirePropertiesAndChild(RequireProperties.followParent(), root);
+        }
+
+        PhysicalProperties properties;
+        if (windowFrameGroup.partitionKeyList.isEmpty()) {
+            properties = new PhysicalProperties(new OrderSpec(requiredOrderKeys));
+        } else {
+            // todo: add new ShuffleType for window, like ShuffleType.WINDOW
+            properties = PhysicalProperties.createHash(
+                windowFrameGroup.partitionKeyList, DistributionSpecHash.ShuffleType.ENFORCED);
+            // requiredOrderKeys contain partitionKeys, so there is no need to check if requiredOrderKeys.isEmpty()
+            properties = properties.withOrderSpec(new OrderSpec(requiredOrderKeys));
```

Comment:
i don't think we should set orderSpec here. Because orderSpec is indicate the order in one `INSTANCE`. but window need order in one `PARTITION`